includes/class-dispatcher.php

```diff
@@ -87,6 +88,48 @@ public function _execute_view( $template ) {
 		return $template;
 	}
 
+    /**
+     * Redirect from draft listings to the main directory page.
+     *
+     * @since x.x.x
+     *
+     * @return void
+     */
+    public function maybe_redirect_from_draft_listings() {
+        global $wp_query, $post, $wpdb;
+
+        if ( $post !== NULL ) {
+            return;
+        }
+
+        $vars = $wp_query->query_vars;
+
+        if ( ! isset( $vars['post_type'] ) || $vars['post_type'] !== WPBDP_POST_TYPE ) {
+            return;
+        }
+
+        // We run a custom query here because we want to check for draft/pending listings and WP_Query doesn't return them.
+        $post = $wpdb->get_results(
```

Comment:
It looks like we might have the rules not mentioning overriding globals. Setting $post here will cause an issue later when we ratchet up the phpcs checks. Can we use a different var name here?

Reply:
Oh 🤦‍♂️ , I didn't realize that I was overwriting the global `$post`, I was too focused on why `get_posts` was not retrieving the post. But I'll answer that in the next comment.